### PR TITLE
Allow calendar selector to default to 12:00 AM on initial click

### PR DESCRIPTION
**Pull request by andalal**
Initially, the calendar selector would default to current time on first date chosen. On subsequent clicks, it would choose 12:00 AM local. Now, the calendar selector always defaults to 12:00 AM local. 

The reason this involves changes in MethodDate is because MethodDate provides a default date. DatePicker passes a value to calendar which could be a MethodDate. If MethodDate returns a default value, originally it was just new Date(). Now, it is new Date() but contains 0, 0, 0 for the hours, minutes, and seconds. 

Tested locally by checking different dates and times.

### CODE REVIEW

File: lib/Common.ts [hunk 1]
```diff
@@ -136,7 +136,14 @@ export class MethodDate {
             }
             this.dateObject.setFullYear(year); // Force 2 digit year to be set correctly
         } else {
+            let today = new Date();
             this.dateObject = new Date();
+            this.dateObject.setFullYear(today.getFullYear());
```
Comment (vishwam) [inline lib/Common.ts]: this seems unnecessary, why do we need this?

Comment (vishwam) [reply lib/Common.ts]: it looks like we just need the setHours/setMinutes/setSeconds

Comment (andalal) [reply lib/Common.ts]: Agreed, removed sets

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # CHANGELOG
 
-## v4.next
+## v4.0.9
 ### Fixed
+- Calendar component would choose current time first time you are selecting a date. Now defaults to 12:00 AM local
```
Comment (Coord) [inline CHANGELOG.md]: I think this change needs to be in a new version now

**Pull request merged**